### PR TITLE
Rewrite verify()

### PR DESCRIPTION
**Pull request by Nekotekina**
Use automatic source location provider intrinsics.

### CODE REVIEW

File: Utilities/Config.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿#include "stdafx.h"
+#include "stdafx.h"
```
Comment (Megamouse) [inline Utilities/Config.cpp:1]: I thought BOM was removed ?

Comment (Nekotekina) [reply Utilities/Config.cpp:1]: You can check yourself

Comment (Nekotekina) [reply Utilities/Config.cpp:1]: I think Utilities/ was untouched

Comment (Megamouse) [reply Utilities/Config.cpp:1]: Ah yes

**Comment by Nekotekina** [comment]
Did some more radical change in the end.

File: Utilities/types.h [hunk 2]
```diff
@@ -904,29 +870,38 @@ struct narrow_impl<From, To, std::void_t<typename From::simple_type>>
...
+	return narrow<u32>(container.size(), line, col, file, func);
 }
 
 // Returns u32 size for an array
 template <typename T, std::size_t Size>
-constexpr u32 size32(const T (&)[Size], const char* msg = nullptr)
+[[nodiscard]] constexpr u32 size32(const T (&)[Size])
 {
+	static_assert(Size < UINT32_MAX, "Array is too big for 32-bit");
 	return static_cast<u32>(Size);
```
Comment (elad335) [inline Utilities/types.h:905]: return u32{Size};

**Pull request merged**